make: Remove -y flag for addlicense

pingleig opened the pull request:
**Description:** <Describe what has changed.>

When using `-y` it is expecting next arg is year, so it was taking `-c` as year and `OpenTelemetry Authors' as input file and end up having error like the following:

```
make addlicense                                                                                                                                                                                          15:23:58
addlicense FAILED => add License errors:

2021/03/15 15:24:10 OpenTelemetry Authors error: lstat OpenTelemetry Authors: no such file or directory

make: *** [addlicense] Error 1
```

The `-y` flag is a string flag and default value is current year.

```go
	year      = flag.String("y", fmt.Sprint(time.Now().Year()), "copyright year(s)")
```


**Link to tracking Issue:** <Issue number if applicable>

N/A

**Testing:** <Describe what testing was performed and which tests were added.>

```
make addlicense                                                                                                                                                                       
Add License finished successfully
```

**Documentation:** <Describe the documentation added.>

pingleig committed:
make: Remove -y flag for addlicense

When using `-y` it is expecting next arg is year, so it was taking `-c`
as year and `OpenTelemetry Authors' as input file and end up having
error like the following:

```
make addlicense                                                                                                                                                                                          15:23:58
addlicense FAILED => add License errors:

2021/03/15 15:24:10 OpenTelemetry Authors error: lstat OpenTelemetry Authors: no such file or directory

make: *** [addlicense] Error 1
```

The `-y` flag is a string flag and default value is current year.

```go
	year      = flag.String("y", fmt.Sprint(time.Now().Year()), "copyright year(s)")
```

djaglowski reviewed (approved):
Confirmed the problem and fix locally.

**Before fix:** 
```
> make addlicense
addlicense FAILED => add License errors:

2021/03/16 09:29:42 OpenTelemetry Authors error: lstat OpenTelemetry Authors: no such file or directory

make: *** [addlicense] Error 1
```

```
// Copyright -c Google LLC
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//      http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
```

**After Fix:**
```
> make addlicense
Add License finished successfully
```

```
// Copyright 2021 OpenTelemetry Authors
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//      http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
```

bogdandrutu commented on Makefile.Common:
```diff
@@ -62,7 +62,7 @@ benchmark:
 
 .PHONY: addlicense
 addlicense:
-	@ADDLICENCESEOUT=`$(ADDLICENCESE) -y -c 'OpenTelemetry Authors' $(ALL_SRC) 2>&1`; \
```
We don't want to use year in our files, so I suggest to do the same as in core https://github.com/open-telemetry/opentelemetry-collector/blob/main/Makefile.Common#L37 and add an empty string after `-y "" -c`

pingleig replied:
Add the empty string, though it seems it with end up having two space between copyright and the holder.  Oops, you already have issue for it https://github.com/google/addlicense/issues/36

pingleig committed:
make: Run addlicense with empty year

Be consistent with core collector per comment
https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/2697#discussion_r595247615
